### DISCUSSION TITLE
formState and proxyFormState objects

### DISCUSSION THREAD

**Discussion by mh-alahdadian** [react-hook-form/react-hook-form#5993] [RFC]
Hi I was trying to remove dependency of Proxy api using js basic feature to set getter for object with known field names using defineProperty
But I didn't understand what are purpose of `_proxyFormState` and `_formState` fields of control object

Can anyone describe it to me ? 

@bluebill1049

**Comment by bluebill1049** [comment]
The proxy is made to knowing which form state is users subscribed. This can lead to much less computation and renders. Eg if ur not subscribed to valid, then its saving all those validation logics.

**Comment by bluebill1049** [comment]
_formState should be able to be removed after react 18 is released with batching update. This is so we can batch a single form state update.

**Comment by mh-alahdadian** [comment]
I just create the PR for removing Proxy #5999

**Reply by bluebill1049** [reply]
Nice. merged.
